vec3.h

```diff
@@ -0,0 +1,120 @@
+//code by Ameya Vikrama Singh
+#include<iostream>
+#include<cmath>
+class vec3{
+    public:
+        float x,y,z;
+
+        vec3(){
+            x=y=z=0;
+        }
+
+        //Cartesian coordinate Constructor
+        vec3(float x,float y,float z){
+            this->x=x;
+            this->y=y;
+            this->z=z;
+        }
+        
+        //assignment operator
+        void operator=(vec3 s){
+            x=s.x;
+            y=s.y;
+            z=s.z;
+        }
+
+        //norm function
+        float norm(){
+            return sqrt(x*x+y*y+z*z);
+        }
+
+        //vector addition
+        vec3 operator+(vec3 s){
```

Comment:
It would also be nice to have `+=`, `-=`, `*=`, `/=`.

Ideally, what you want to do for `+`, `-`, `*`, and `/` is that you want to overload the respective compound assignment operators (aka `+=`, `-=`, `*=`, `/=`), and then define these ones in terms of those.

Doing this kind of bootstrapping reduces the amount of changes you have to make when you want to change some logic, and also makes the code a bit cleaner (in this case).

Reply:
Another thing you need to do is make the `s` variable `const`. Refer to [this](https://gist.github.com/beached/38a4ae52fcadfab68cb6de05403fa393#operator-addition) for the correct function signature.

(Make sure to fix all the operators!)